thanks/thanks.py

```diff
@@ -1,82 +1,97 @@
 # -*- coding: utf-8 -*-
 
+from collections import namedtuple
 from humanfriendly.tables import format_pretty_table
 import json
 import os
-try:
-    import pip._internal as pip
-except ImportError:
-    import pip
 import requirements
+import requests
 from termcolor import colored, cprint
-try:
-    import xmlrpclib
-except ImportError:
-    import xmlrpc.client as xmlrpclib
 
 
-json_file = ("{}/thanks.json".format(os.path.dirname(os.path.realpath(__file__))))
-client = xmlrpclib.ServerProxy('https://pypi.python.org/pypi')
+JSON_FILE = ("{}/thanks.json".format(os.path.dirname(os.path.realpath(__file__))))
+
+ProjectData = namedtuple('ProjectData', ['name', 'funding_link', 'authors'])
 
 
 class Thanks():
     def __init__(self, debug=False):
         self.debug = debug
-        self.data = {}
-        with open(json_file, 'r') as j:
-            print('Loading data about {}'.format(colored('contributors...', 'cyan')))
-            self.data = json.load(j)
+        self.data = self.load_local_project_data()
         self.give_thanks_to = {}
-        self.local_installs = {}
-        for package in pip.get_installed_distributions():
-            self.local_installs[package.key] = package
 
-    def find_package_roles(self, requirements_file):
-        with open(requirements_file, 'r') as reqs:
-            print('Scanning your {} file...'.format(colored('requirements', 'red')))
-            for req in requirements.parse(reqs):
-                self.update_give_thanks(req.name)
-            if req.name in self.local_installs:
-                for package in self.local_installs[req.name].requires():
-                    self.update_give_thanks(package.key)
-            if self.give_thanks_to:
-                cprint(
-                    'You depend on {} who would {}'.format(
-                        colored('{} authors'.format(len(self.give_thanks_to)), 'cyan'),
-                        colored('enjoy donations!', 'green'),
-                    ), attrs=['bold'])
-                table_data = []
-                for contributor in self.give_thanks_to:
-                    table_data.append(
-                        [
-                            colored(contributor, 'cyan'),
-                            colored(self.give_thanks_to[contributor]['url'], 'green'),
-                            colored(", ".join(set(self.give_thanks_to[contributor]['packages'])), 'red'),
-                        ]
-                    )
-                print(format_pretty_table(
-                    data=table_data,
-                    column_names=['Author', 'Where to thank', 'Packages'],
-                    horizontal_bar=' ',
-                    vertical_bar=' ',
-                ))
+    def load_local_project_data(self):
```

Comment:
I'm confused why we're doing this when we have the lovely `package_tools` module you wrote.

Reply:
I simply didn't want to do too much in a single commit/PR.
Refactor was also to aid in describing what all the parts of `Thanks` were doing and help me visualise what I would be changing when I started using the `package_tools` module.